internal/spec/build.go

```diff
@@ -27,14 +28,14 @@ func FromScript(ctx context.Context, script string, now time.Time) (*flux.Spec,
 		return nil, err
 	}
 	s.Finish()
-	return FromAST(ctx, astPkg, now)
+	return FromAST(ctx, nil, astPkg, now)
```

Comment:
verify we don't need to pass `deps` through here

Reply:
We do need the deps here, simplest way is to require them to be passed in.